### PR TITLE
chore: Adds staticcheck lint check

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

Adds the `staticcheck` lint check and makes the necessary change to have it pass.

### CODE REVIEW

**Comment by gab-arrobo** [comment]
ok to test

File: context/management_data.go [hunk 1]
```diff
@@ -56,7 +55,6 @@ func NnrfNFManagementDataModel(nf *models.NfProfile, nfprofile models.NfProfile)
 }
 
 func SetsubscriptionId() string {
-	rand.Seed(time.Now().UnixNano())
```
Comment (gab-arrobo) [inline context/management_data.go:59]: Same comment as here: https://github.com/omec-project/ausf/pull/37#discussion_r1442014164

Comment (gruyaume) [reply context/management_data.go:59]: This was addressed in the last commit

**Comment by gab-arrobo** [comment]
@gruyaume, I will try to continue reviewing these PRs later tonight.

**Review by gab-arrobo — APPROVED** [review]
+1 (tested changes with AiaB)

**Pull request merged**